diplom.py

```diff
@@ -9,6 +9,25 @@ def parse(line):
     label = int (tokens[23])
     return (features, label)
 
+def mean_average_precision (predicted_probability_distribution, 
+    n_of_classes, 
```

Comment:
Also, do you need dataset_size? I think you can derive it from the predicted_probability_distribution i.e. len(predicted_probability_distribution).